executor/pkg_test.go

```diff
@@ -111,6 +117,102 @@ func (s *pkgTestSuite) TestNestedLoopApply(c *C) {
 	}
 }
 
+func prepareOneColChildExec(sctx sessionctx.Context, rowCount int) Executor {
+	col0 := &expression.Column{Index: 0, RetType: types.NewFieldType(mysql.TypeLong)}
+	schema := expression.NewSchema(col0)
+	exec := &MockExec{
+		baseExecutor: newBaseExecutor(sctx, schema, ""),
+		Rows:         make([]chunk.MutRow, rowCount)}
+	for i := 0; i < len(exec.Rows); i++ {
+		exec.Rows[i] = chunk.MutRowFromDatums(types.MakeDatums(i % 10))
+	}
+	return exec
+}
+
+func prepare4RadixPartition(sctx sessionctx.Context, rowCount int) *HashJoinExec {
+	childExec0 := prepareOneColChildExec(sctx, rowCount)
+	childExec1 := prepareOneColChildExec(sctx, rowCount)
+
+	col0 := &expression.Column{Index: 0, RetType: types.NewFieldType(mysql.TypeLong)}
+	col1 := &expression.Column{Index: 0, RetType: types.NewFieldType(mysql.TypeLong)}
+	joinSchema := expression.NewSchema(col0, col1)
+	hashJoinExec := &HashJoinExec{
+		baseExecutor:   newBaseExecutor(sctx, joinSchema, "HashJoin", childExec0, childExec1),
+		concurrency:    4,
+		joinType:       0, // InnerJoin
+		innerIdx:       0,
+		innerKeys:      []*expression.Column{{Index: 0, RetType: types.NewFieldType(mysql.TypeLong)}},
+		innerKeyColIdx: []int{0},
+		outerKeys:      []*expression.Column{{Index: 0, RetType: types.NewFieldType(mysql.TypeLong)}},
+		outerKeyColIdx: []int{0},
+		innerExec:      childExec0,
+		outerExec:      childExec1,
+	}
+	return hashJoinExec
+}
+
+func (s *pkgTestSuite) TestRadixPartition(c *C) {
+	sctx := mock.NewContext()
+	hashJoinExec := prepare4RadixPartition(sctx, 200)
+	sv := sctx.GetSessionVars()
+	originL2CacheSize, originEnableRadixJoin, originMaxChunkSize := sv.L2CacheSize, sv.EnableRadixJoin, sv.MaxChunkSize
+	sv.L2CacheSize = 100
+	sv.EnableRadixJoin = true
+	sv.MaxChunkSize = 100
+	defer func() {
+		sv.L2CacheSize, sv.EnableRadixJoin, sv.MaxChunkSize = originL2CacheSize, originEnableRadixJoin, originMaxChunkSize
+	}()
+	sv.StmtCtx.MemTracker = memory.NewTracker("RootMemTracker", variable.DefTiDBMemQuotaHashJoin)
+
+	ctx := context.Background()
+	err := hashJoinExec.Open(ctx)
+	c.Assert(err, IsNil)
+
+	innerResultCh := make(chan *chunk.Chunk, hashJoinExec.concurrency)
```

Comment:
We have changed the channel capacity from `e.concurrency` to `1` in `fetchInnerAndBuildHashTable`, so we should use `1` here also?